src/main/java/com/epam/ta/reportportal/dao/TestItemRepositoryCustom.java

```diff
@@ -192,4 +192,30 @@
 	 */
 	TestItemTypeEnum getTypeByItemId(Long itemId);
 
+	/**
+	 * Select item IDs by launch ID and issue type ID with logs which level is greater than or equal to provided
+	 * and message is matched by the STRING pattern
+	 *
+	 * @param launchId    {@link com.epam.ta.reportportal.entity.launch.Launch#id}
+	 * @param issueTypeId {@link IssueType#id}
+	 * @param logLevel    {@link com.epam.ta.reportportal.entity.log.Log#logLevel}
+	 * @param pattern     CASE SENSITIVE STRING pattern for log message search
+	 * @return The {@link List} of the {@link TestItem#itemId}
+	 */
+	List<Long> selectIdsByLaunchIdAndIssueTypeIdAndLogLevelAndLogMessageStringPattern(Long launchId, Long issueTypeId, Integer logLevel,
+			String pattern);
+
+	/**
+	 * Select item IDs by launch ID and issue type ID with logs which level is greater than or equal to provided
+	 * and message is matched by the REGEX pattern
+	 *
+	 * @param launchId    {@link com.epam.ta.reportportal.entity.launch.Launch#id}
+	 * @param issueTypeId {@link IssueType#id}
+	 * @param logLevel    {@link com.epam.ta.reportportal.entity.log.Log#logLevel}
+	 * @param pattern     REGEX pattern for log message search
+	 * @return The {@link List} of the {@link TestItem#itemId}
+	 */
+	List<Long> selectIdsByLaunchIdAndIssueTypeIdAndLogLevelAndLogMessageRegexPattern(Long launchId, Long issueTypeId, Integer logLevel,
```

Comment:
same

we can not write byLaunchId/IssueTypeId etc, because we can recognize it from parameters.
so it's better call it according patter logic.
for example "selectPatternMatchedItems" or smth like that

Reply:
renamed